shared/AppInsightsCore/src/JavaScriptSDK/StatsBeat.ts

```diff
@@ -0,0 +1,198 @@
+import dynamicProto from "@microsoft/dynamicproto-js";
+import { ITimerHandler, scheduleTimeout, utcNow } from "@nevware21/ts-utils";
+import { IAppInsightsCore } from "../JavaScriptSDK.Interfaces/IAppInsightsCore";
+import { IStatsBeat } from "../JavaScriptSDK.Interfaces/IStatsBeat";
+import { ITelemetryItem } from "../JavaScriptSDK.Interfaces/ITelemetryItem";
+import { IPayloadData } from "../JavaScriptSDK.Interfaces/IXHROverride";
+import { NetworkStatsbeat, createNetworkStatsbeat } from "./NetworkStatsbeat";
+
+const INSTRUMENTATION_KEY = "c4a29126-a7cb-47e5-b348-11414998b11e";
+const STATS_COLLECTION_SHORT_INTERVAL: number = 900000; // 15 minutes
+const STATSBEAT_LANGUAGE = "JavaScript";
+const STATSBEAT_TYPE = "Browser";
+
+export class Statsbeat implements IStatsBeat {
+    constructor() {
+        let _networkCounter: NetworkStatsbeat;
+        let _isEnabled: boolean = false;
+        let _core: IAppInsightsCore;
+        let _timeoutHandle: ITimerHandler;      // Handle to the timer for sending telemetry. This way, we would not send telemetry when system sleep.
+        // Custom dimensions
+        let _cikey: string;
+        let _language: string;
+        let _sdkVersion: string;
+        let _os: string;
+        let _runTimeVersion: string;
+        dynamicProto(Statsbeat, this, (_self, _base) => {
+            _self.initialize = (core: IAppInsightsCore, ikey: string, endpoint: string, version?: string) => {
```

Comment:
Lets update the signature to pass the values as a config object (iKey, endpoint, version) as I'm feeling that we might want to update this in the future.

We could also then make these dynamic, so we could "update" then for things like passing in the browser version etc. So basically follow the normal type of "onConfigChange()" with setting defaults -- the code "supports" defining this for internal types (not part of the global config object (IConfiguration)), but it may need some signature updates or we fudge it.

For now just create an interface with these 3 and pass them in (ie. don't treat as dynamic or use the onConfigChange)